db/migrate/20160106102516_extract_estate_to_table.rb

```diff
@@ -0,0 +1,41 @@
+class ExtractEstateToTable < ActiveRecord::Migration
+  def up
+    create_table :estates, id: :uuid do |t|
+      t.string :name, null: false
+      t.timestamps
+    end
+
+    add_index :estates, :name, unique: true
+    add_column :prisons, :estate_id, :uuid
+
+    execute <<-SQL
+      INSERT INTO estates (name)
+      SELECT DISTINCT estate FROM prisons
+    SQL
+
+    execute <<-SQL
+      UPDATE prisons
+      SET estate_id = estates.id
+      FROM estates
+      WHERE estate = estates.name
+    SQL
+
+    change_column_null :prisons, :estate_id, false
+    remove_column :prisons, :estate
+    add_foreign_key :prisons, :estates
```

Comment:
The scenario I envisaged was an entire estate being removed.  I suppose we could deal with that by disabling all the prisons, though.  Ok, I'm satisfied...